himan-scripts/rail_weather.lua

```diff
@@ -0,0 +1,137 @@
+function SumT(TMinus,TPlus,curTime,par)
+  local psum = {}
+
+  local stopStep = math.max(curTime:GetStep():Hours() + TMinus , 0)
+  local startStep = math.min(curTime:GetStep():Hours() + TPlus , 240)
+
+  -- set initial time step
+  local mytime = forecast_time(curTime:GetOriginDateTime(),time_duration(HPTimeResolution.kHourResolution,startStep))
+
+
+  while true do
+
+    if mytime:GetStep():Hours() < stopStep or mytime:GetStep():Hours() < 0 then
+      break
+    end
+
+    local p = luatool:FetchWithType(current_time, current_level, par, current_forecast_type)
+
+    if p then
+      if #psum == 0 then
+        for i=1, #p do
+         psum[i] = 0
+        end
+      end
+
+      for i=1, #p do
+        psum[i] = psum[i] + p[i] 
+      end
+    end
+
+    -- adjust time
+    mytime:GetValidDateTime():Adjust(HPTimeResolution.kHourResolution, -1)
+
+  end
+
+  return psum
+end
+
+function MinT(TMinus,TPlus,curTime,par)
+  local pmin = {}
+
+  local stopStep = math.max(curTime:GetStep():Hours() + TMinus , 0)
+  local startStep = math.min(curTime:GetStep():Hours() + TPlus , 240)
+
+  -- set initial time step
+  local mytime = forecast_time(curTime:GetOriginDateTime(),time_duration(HPTimeResolution.kHourResolution,startStep))
+
+  while true do
+
+    if mytime:GetStep():Hours() < stopStep or mytime:GetStep():Hours() < 0 then
+      break
+    end
+
+    local p = luatool:FetchWithType(mytime, current_level, par, current_forecast_type)
+
+    if p then
+      if #pmin == 0 then
+        for i=1, #p do
+         pmin[i] = 1e38
+        end
+      end
+
+      for i=1, #p do
+        pmin[i] = math.min(pmin[i],p[i])
+      end
+    end
+
+    -- adjust time
+    mytime:GetValidDateTime():Adjust(HPTimeResolution.kHourResolution, -1)
+
+  end
+
+  return pmin
+end
+
+function MaxT(TMinus,TPlus,curTime,par,curLevel)
```

Comment:
What does the "T" in the function name signal?

Reply:
Time I assume. That's how they are named in smartmet macro.